pkg/operator/operator.go

```diff
@@ -101,9 +108,53 @@ func RunOperator(ctx context.Context, controllerConfig *controllercmd.Controller
 	o.StatusManager = statusmanager.New(o.client, "network", cluster)
 	defer utilruntime.HandleCrash(o.StatusManager.SetDegradedOnPanicAndCrash)
 
+	klog.Infof("Fetching cluster feature gates...")
+	kubeConfig := o.client.Default().Config()
+	kubeClient := o.client.Default().Kubernetes()
+	configClient, err := configclient.NewForConfig(kubeConfig)
+	if err != nil {
+		return err
+	}
+	configInformers := configinformers.NewSharedInformerFactory(configClient, 10*time.Minute)
+	desiredVersion := os.Getenv("RELEASE_VERSION")
+	missingVersion := "0.0.1-snapshot"
+
+	eventRecorder := events.NewKubeRecorder(kubeClient.CoreV1().Events("openshift-network-operator"), "cluster-network-operator", &corev1.ObjectReference{
+		APIVersion: "apps/v1",
+		Kind:       "Deployment",
+		Namespace:  "openshift-network-operator",
+		Name:       "network-operator",
+	}, clock.RealClock{})
+
+	featureGateAccessor := featuregates.NewFeatureGateAccess(
+		desiredVersion, missingVersion,
+		configInformers.Config().V1().ClusterVersions(), configInformers.Config().V1().FeatureGates(),
+		eventRecorder,
```

Comment:
can we use `controllerConfig.EventRecorder` instead?

Reply:
Ah, yes, sure